src/pykeen/models/base.py

```diff
@@ -490,6 +611,195 @@ def predict_scores_all_heads(
             scores = torch.sigmoid(scores)
         return scores
 
+    def _score_all_triples(
+        self,
+        batch_size: int = 1,
+        return_tensors: bool = False,
+        *,
+        add_novelties: bool = True,
+        remove_known: bool = False,
+        testing: Optional[torch.LongTensor] = None,
+    ) -> Union[Tuple[torch.LongTensor, torch.FloatTensor], pd.DataFrame]:
+        """Compute and store scores for all triples.
+
+        :return: Parallel arrays of triples and scores
+        """
+        # initialize buffer on cpu
+        scores = torch.empty(self.num_relations, self.num_entities, self.num_entities, dtype=torch.float32)
```

Comment:
should we specify the device here?

Reply:
From the [documentation](https://pytorch.org/docs/stable/generated/torch.empty.html?highlight=empty#torch.empty)

``` python
torch.empty(*size, out=None, dtype=None, layout=torch.strided, device=None, requires_grad=False, pin_memory=False) → Tensor
```

and 

> `device (torch.device, optional)` – the desired device of returned tensor. Default: if `None`, uses the current device for the default tensor type (see `torch.set_default_tensor_type()`). device will be the CPU for CPU tensor types and the current CUDA device for CUDA tensor types.

So setting it to `cpu` should be the safer option.